### PR TITLE
prov/gni: Fix scalable ep merge issue

### PR DESCRIPTION
**Pull request by chuckfossen**
Fixes ofi-cray/libfabric-cray#1052

Signed-off-by: Chuck Fossen <chuckf@cray.com>

@sungeunchoi

### CODE REVIEW

File: prov/gni/test/sep.c [hunk 1]
```diff
@@ -230,6 +227,9 @@ void sep_setup(void)
 					 FI_RECV | FI_READ);
 			cr_assert(!ret, "fi_ep_bind");
 
+			ret = fi_enable(tx_ep[i][j]);
+			cr_assert(!ret, "fi_enable");
+
```
Comment (sungeunchoi) [inline prov/gni/test/sep.c:232]: I'm confused by this.  Why does enabling tx_ep[i][j] have to come after binding to rx_ep[i][j]?

Comment (hppritcha) [reply prov/gni/test/sep.c:232]: because that's the rule (see the fi_endpoint man page)

**Comment by hppritcha** [comment]
I don't think this is the right fix.  I don't entirely like it, but I think we need to modify the check for enabled state in gnix_ep_bind to check if both tx_enabled and rx_enabled have been set, so &&
not ||.  I'd put a comment in the area explaining we do this because for SEP's we use a single ```gnix_fid_ep``` for the tx and rx contexts.

File: prov/gni/test/sep.c (hunk 1 above)
Comment (chuckfossen) [reply prov/gni/test/sep.c:232]: We are actually binding to the gnix_ep that is contained in the tx/rx context (which are sharing the the gnix_ep).

**Comment by chuckfossen** [comment]
@hppritcha, are you saying that both rx_enabled and tx_enabled need to be set for the check to fail?

**Comment by chuckfossen** [comment]
Closing as we decided to fix the check in ep bind.

**Pull request closed without merging**